go-controller/pkg/ovn/gress_policy.go

```diff
@@ -0,0 +1,369 @@
+	if !gp.peerAddressSets[hashedAddressSet] {
+		return "", "", false
+	}
+
+	oldL3Match := gp.getL3MatchFromAddressSet()
+
+	for i, addressSet := range gp.sortedPeerAddressSets {
+		if addressSet == hashedAddressSet {
+			gp.sortedPeerAddressSets = append(
+				gp.sortedPeerAddressSets[:i],
+				gp.sortedPeerAddressSets[i+1:]...)
+			break
+		}
+	}
+	delete(gp.peerAddressSets, hashedAddressSet)
+
+	return oldL3Match, gp.getL3MatchFromAddressSet(), true
+}
+
+func (gp *gressPolicy) localPodAddACL(portGroupName, portGroupUUID string) {
+	l3Match := gp.getL3MatchFromAddressSet()
+
+	var lportMatch, cidrMatch string
+	if gp.policyType == knet.PolicyTypeIngress {
+		lportMatch = fmt.Sprintf("outport == @%s", portGroupName)
+	} else {
+		lportMatch = fmt.Sprintf("inport == @%s", portGroupName)
+	}
+
+	// If IPBlock CIDR is not empty and except string [] is not empty,
+	// add deny acl rule with priority ipBlockDenyPriority (1010).
+	if len(gp.ipBlockCidr) > 0 && len(gp.ipBlockExcept) > 0 {
+		except := fmt.Sprintf("{%s}", strings.Join(gp.ipBlockExcept, ", "))
+		if err := gp.addIPBlockACLDeny(except, ipBlockDenyPriority, portGroupName, portGroupUUID); err != nil {
+			klog.Warningf(err.Error())
+		}
+	}
+
+	if len(gp.portPolicies) == 0 {
+		match := fmt.Sprintf("match=\"%s && %s\"", l3Match, lportMatch)
+		l4Match := noneMatch
+
+		if len(gp.ipBlockCidr) > 0 {
+			// Add ACL allow rule for IPBlock CIDR
+			cidrMatch = gp.getMatchFromIPBlock(lportMatch, l4Match)
+			if err := gp.addACLAllow(cidrMatch, l4Match, portGroupUUID, true); err != nil {
+				klog.Warningf(err.Error())
+			}
+		}
+		// if there are pod/namespace selector, then allow packets from/to that address_set or
+		// if the NetworkPolicyPeer is empty, then allow from all sources or to all destinations.
+		if len(gp.sortedPeerAddressSets) > 0 || len(gp.ipBlockCidr) == 0 {
+			if err := gp.addACLAllow(match, l4Match, portGroupUUID, false); err != nil {
+				klog.Warningf(err.Error())
+			}
+		}
+	}
+	for _, port := range gp.portPolicies {
+		l4Match, err := port.getL4Match()
+		if err != nil {
+			continue
+		}
+		match := fmt.Sprintf("match=\"%s && %s && %s\"", l3Match, l4Match, lportMatch)
+		if len(gp.ipBlockCidr) > 0 {
+			// Add ACL allow rule for IPBlock CIDR
+			cidrMatch = gp.getMatchFromIPBlock(lportMatch, l4Match)
+			if err := gp.addACLAllow(cidrMatch, l4Match, portGroupUUID, true); err != nil {
+				klog.Warningf(err.Error())
+			}
+		}
+		if len(gp.sortedPeerAddressSets) > 0 || len(gp.ipBlockCidr) == 0 {
+			if err := gp.addACLAllow(match, l4Match, portGroupUUID, false); err != nil {
+				klog.Warningf(err.Error())
+			}
+		}
+	}
+}
+
+func (gp *gressPolicy) addACLAllow(match, l4Match, portGroupUUID string, ipBlockCidr bool) error {
+	var direction, action string
+	direction = toLport
+	if gp.policyType == knet.PolicyTypeIngress {
+		action = "allow-related"
+	} else {
+		action = "allow"
+	}
+
+	uuid, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
```

Comment:
Any reason not to just pass `--may-exist` and issue a single command?

Reply:
@squeed it's a C&P of the exsiting code so I didn't modify the logic here.